src/RegistrationController.js

```diff
@@ -128,15 +128,18 @@ function (
       });
       this.model = new Model();
 
-      var checkPasswordMeetComplexities = function(model) {
+      var checkPasswordMeetComplexities = function (model, showError) {
         var password = model.get('password') || '';
-        _.each(schema.passwordComplexity.enabledComplexities, function(complexityName) {
-          var ele = Okta.$('.password-complexity-' + complexityName);
+        _.each(schema.passwordComplexity.enabledComplexities, function (complexityName) {
+          var ele = Okta.$('#password-complexity-' + complexityName);
           var complexityValue = this.get(complexityName);
-          if (PasswordComplexityUtil.complexities[complexityName].doesComplexityMeet(complexityValue, password, model)){
-            ele.addClass('password-complexity-meet');
+          ele.removeClass('password-complexity-satisfied password-complexity-unsatisfied password-complexity-error');
+          if (PasswordComplexityUtil.complexities[complexityName].doesComplexityMeet(complexityValue, password, model)) {
+            ele.addClass('password-complexity-satisfied');
+          } else  if (showError) {
```

Comment:
Nit: ``} else if (showError) {`` (extra space between the ``else`` and the ``if``)